src/commands/mixu.ts

```diff
@@ -1,13 +1,47 @@
 import { Guild } from 'discord.js'
 import { Discord, SimpleCommand, SimpleCommandMessage } from 'discordx'
+import { injectable } from 'tsyringe'
+import { ORM } from '../persistence'
+
+interface BestMixu {
+  owner: string
+  tiles: number[]
+  score: number
+}
 
 @Discord()
-abstract class Mixu {
+@injectable()
+class Mixu {
   private numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
-  private bestScore = 0
-  private bestMixuTiles: number[] = []
+  private queried = false
+  private bestMixu: BestMixu = {
+    owner: '',
+    tiles: [],
+    score: 0,
+  }
+
   private mixuChannel = '340275382093611011'
 
+  constructor(private client: ORM) {}
```

Comment:
I think it might be a good idea to set bestMixu in the constructor instead of having this queried state in the findBestMixu function.

Reply:
That was my first intention but I ran into issues because of the construct not allowing async calls, I'm not really sure what is the best way to get around that.

Reply:
Oh, you might be right, your method should work better then.